src/hera/workflows/script.py

```diff
@@ -555,8 +555,11 @@ def _extract_return_annotation_output(source: Callable) -> List:
     if get_workflow_annotation(return_annotation):
         output.append(annotation_args)
     elif origin_type is tuple:
-        for annotated_type in annotation_args:
-            output.append(get_args(annotated_type))
+        if all(get_workflow_annotation(annotated_type) for annotated_type in annotation_args):
+            for annotated_type in annotation_args:
+                output.append(get_args(annotated_type))
+        elif any(get_workflow_annotation(annotated_type) for annotated_type in annotation_args):
```

Comment:
nit: more DRY to do the transformation loop only once? perhaps
```suggestion
        num_workflow_annotations = sum(1 for annotated_type in annotation_args if get_workflow_annotation(annotated_type))
        if len(annotation_args) == num_workflow_annotations:
            output.extend(get_args(annotated_type) for annotated_type in annotation_args)
        elif num_workflow_annotations > 0:
```

Reply:
This is IMO harder to understand than the use of `all` and `any` - happy to save us doing the transformation twice but could we add a comment to explain?